docs/changes/newsfragments/7012.improved

```diff
@@ -0,0 +1,2 @@
+Save the parameter snapshot under the long name key in the parameters snapshot in datasets (not the station snapshot), to avoid overwritting the snapshot multiple parameters have identical short names.
```

Comment:
```suggestion
Save the parameter snapshot under the `register_name` key in the parameters snapshot in datasets (not the station snapshot), to avoid overwritting the snapshot multiple parameters have identical `short_name`s.
```